rfcs/proposed/host_backend_histogram/README.md

```diff
@@ -0,0 +1,167 @@
+# Host Backends Support for the Histogram APIs
+
+## Introduction
+The oneDPL library added histogram APIs, currently implemented only for device policies with the DPC++ backend. These
+APIs are defined in the oneAPI Specification 1.4. Please see the
+[oneAPI Specification](https://github.com/uxlfoundation/oneAPI-spec/blob/main/source/elements/oneDPL/source/parallel_api/algorithms.rst#parallel-algorithms)
+for the details. The host-side backends (serial, TBB, OpenMP) are not yet supported. This RFC proposes extending
+histogram support to these backends.
+
+The pull request for the proposed implementation exists [here](https://github.com/oneapi-src/oneDPL/pull/1974).
+
+## Motivations
+There are many cases to use a host-side serial or a host-side implementation of histogram. Another motivation for adding
+the support is simply to be spec compliant with the oneAPI specification.
+
+## Design Considerations
+
+### Key Requirements
+Provide support for the `histogram` APIs with the following policies and backends:
+- Policies: `seq`, `unseq`, `par`, `par_unseq`
+- Backends: `serial`, `tbb`, `openmp`
+
+Users have a choice of execution policies when calling oneDPL APIs. They also have a number of options of backends
+which they can select from when using oneDPL. It is important that all combinations of these options have support for
+the `histogram` APIs.
+
+### Performance
+With little computation, a histogram algorithm is likely a memory-bound algorithm. So, the implementation prioritize
+reducing memory accesses and minimizing temporary memory traffic.
+
+### Memory Footprint
```

Comment:
Do we wish to specify space / computational complexity of the proposed algorithm somewhere, or is this too much?

Reply:
I've added some sections on computation complexity and temporary memory requirements.

I don't think we should add something like this to the specification though, because I don't want to dictate the algorithm used from the specification, but its good to document it in the current proposal.  Thanks.